core/src/main/java/org/apache/druid/java/util/common/NonnullPair.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.java.util.common;
+
+import com.google.common.base.Preconditions;
+
+public class NonnullPair<L, R> extends Pair<L, R>
+{
+  public NonnullPair(L lhs, R rhs)
+  {
+    super(Preconditions.checkNotNull(lhs, "lhs"), Preconditions.checkNotNull(rhs, "rhs"));
+  }
+
+  public L left()
```

Comment:
I think it would be better to encapsulate them in `Pair` in terms of OOP (maybe aesthetically better too), but I don't really have a strong preference here since both fields in `Pair` are final. `Pair` is already in use everywhere so I changed `NonnullPair` to be consistent.